### PR TITLE
Fix compiler warnings.  No functional changes.

### PR DESCRIPTION
**Pull request by RichardBarry**
These updates fix all the warnings other than those that require an update to the MQTT agent sub-module.  That sub-module has a mismatch in the const qualification of the first parameter in the following two prototypes:

typedef bool ( * MQTTAgentMessageSend_t )( MQTTAgentMessageContext_t * pMsgCtx,
                                           MQTTAgentCommand_t * const * pCommandToSend,
                                           uint32_t blockTimeMs );

typedef bool ( * MQTTAgentMessageRecv_t )( MQTTAgentMessageContext_t * pMsgCtx,
                                           MQTTAgentCommand_t ** pReceivedCommand,
                                           uint32_t blockTimeMs );

To fix those warnings add a const to the first parameter as below, or update the code using the prototypes to remove the const if adding the cost risks breaking other code:

typedef bool ( * MQTTAgentMessageSend_t )( **const** MQTTAgentMessageContext_t * pMsgCtx,
                                           MQTTAgentCommand_t * const * pCommandToSend,
                                           uint32_t blockTimeMs );

typedef bool ( * MQTTAgentMessageRecv_t )( **const** MQTTAgentMessageContext_t * pMsgCtx,
                                           MQTTAgentCommand_t ** pReceivedCommand,
                                           uint32_t blockTimeMs );

### CODE REVIEW

**Review by abhidixi11 — APPROVED** [review]

File: build/Cortex-M3_MPS2_QEMU_GCC/target-specific-source/startup.c [hunk 1]
```diff
@@ -74,6 +119,9 @@ volatile uint32_t lr; /* Link register. */
 volatile uint32_t pc; /* Program counter. */
 volatile uint32_t psr;/* Program status register. */
 
+/* Called from the hardfault handler to provide information on the processor
+ * state at the time of the faul.
```
Comment (aggarw13) [inline build/Cortex-M3_MPS2_QEMU_GCC/target-specific-source/startup.c:123]: Typo
```suggestion
 * state at the time of the fault.
```

File: build/Cortex-M3_MPS2_QEMU_GCC/target-specific-source/startup.c [hunk 2]
```diff
@@ -92,26 +140,23 @@ void prvGetRegistersFromStack( uint32_t *pulFaultStackAddress )
...
-        "Infinite_Loop:\n"
-        "    b  Infinite_Loop\n"
-        ".size  Default_Handler, .-Default_Handler\n"
-        ".align 4\n"
-        "NVIC_INT_CTRL_CONST: .word 0xe000ed04\n"
+        " ldr r3, NVIC_INT_CTRL_CONST            \n" /* Load the address of the interrupt control register into r3. */
+        " ldr r2, [r3, #0]                       \n" /* Load the value of the interrupt control register into r2. */
+        " uxtb r2, r2                            \n" /* The interrupt number is in the least significant byte - clear all other bits. */
+        "Infinite_Loop:                          \n" /* Sit in an infinite loop - the number of the executing interrupt is held in r2. */
+        " b  Infinite_Loop                       \n"
```
Comment (aggarw13) [inline build/Cortex-M3_MPS2_QEMU_GCC/target-specific-source/startup.c:152]: The change removes the ".size Default_Handler, .-Default_Handler\n" line. Is it intentional?

**Comment by muneebahmed10** [comment]
> To fix those warnings add a const to the first parameter as below, or update the code using the prototypes to remove the const if adding the cost risks breaking other code

I think the `const` should be removed from the implementation here. If the agent was used with some non-FreeRTOS implementation then sending to a queue might require a non-`const` context parameter

**Comment by RichardBarry** [comment]
As per your suggestion, I will resolve the last remaining compiler warnings by removing the const from the app code so the library remains unchanged.

**Review by muneebahmed10 — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**